fix, author dropdown positioning, refs steemit/condenser #1533

gl2748 opened the pull request:
# Before:
![before](https://user-images.githubusercontent.com/3374538/27773270-e6687fa0-5f43-11e7-9575-46cae846b8c2.gif)

# After:
![after](https://user-images.githubusercontent.com/3374538/27773272-ebef115a-5f43-11e7-8034-f48bd1a035f9.gif)

gl2748 commented:
Things that still need to be done.

- [ ] Strip react-foundation-components dependency from here: https://github.com/steemit/condenser/blob/master/app/components/pages/UserProfile.jsx#L22

- [ ] Need to make the overlay close on any-click outside the overlay div element. Like this:
https://github.com/steemit/condenser/blob/master/app/components/elements/FoundationDropdown.jsx#L37-L44

sneak commented:
Lookin good.  We have this same issue with the user action menu dropdown on mobile.

sneak commented:
FYI we use four-space indents

valzav commented:
@gl2748, great work! Any chance you could refactor this into a separate component so we could replace LinkWithDropdown with it in other places?

gl2748 commented:
@sneak @valzav,
Yes, I'll work on this over the weekend and revise the PR accordingly.
Thanks for all your feedback.
Iain.

roadscape commented:
@gl2748 are you still willing to make these changes or should we merge as-is?

gl2748 commented:
@roadscape yes! I'm going to chat with Ned today at somepoint. I'll keep you posted.

gl2748 commented:
@valzav @roadscape,

I have been working through this.  Please have a look at my approach. I am not sure of the need for a new component, the `Overlay `component offered by `react-overlays` is fairly powerful by itself. Info here:
https://react-bootstrap.github.io/react-overlays/#overlay

What I have done is taken the author dropdown and turned it into a stateless-functional-component (SFC). Perhaps this is the best way forward:

Basically:
Using the Overlay component where necessary and passing it the appropriate contents as a `SFC`.

If you agree, I can continue this line of work refactoring components to remove dependencies on the `react-foundation-components`  next up would be the `FoundationDropdown` component ...

gl2748 commented:
Today I have moved onto removing `react-foundation-components` dependency from the `Author` page.

roadscape commented:
@gl2748 looking good. Could you please look into why the styles changed?

Before:
![image](https://user-images.githubusercontent.com/5168676/28578638-50434cca-7128-11e7-935c-9ffdefe2cf55.png)

After:
![image](https://user-images.githubusercontent.com/5168676/28578656-5b19afa4-7128-11e7-83c9-03b9506397b3.png)

roadscape commented:
Also could you please rebase on master?

gl2748 commented:
@roadscape I've resolved those issues with the style creeping, rebased on master and split out further work on removing react-foundation-components  into another branch that'll I'll have a PR for shortly. Screen caps below for the updates to match Prod.
### Before.
![steembefore](https://user-images.githubusercontent.com/3374538/28616277-2e9b4530-71ca-11e7-8936-a5a66fb2f799.png)

### After.
![steemafter](https://user-images.githubusercontent.com/3374538/28616281-33563c56-71ca-11e7-9719-393ebaadcb07.png)

roadscape commented:
Wonderful, thanks. Be aware we have an unusually large set of changes to merge into master soon (`rc` branch). I might merge this PR first and fix conflicts in the other branch.

roadscape commented on app/components/elements/Icon.scss:
```diff
@@ -4,7 +4,6 @@
   > svg, span.icon {
     width: 1.12rem;
     height: 1.12rem;
-    vertical-align: top;
```
Are we sure it's safe to get rid of this?

gl2748 replied:
@roadscape,  Yes, I am confident it is safe to remove that.
